src/coprocessor/statistics/analyze.rs

```diff
@@ -219,36 +261,57 @@ impl<S: Snapshot> SampleBuilder<S> {
             );
             self.col_len
         ];
-        while let Some(row) = self.data.next()? {
-            let row = row.take_origin()?;
-            let cols = row.get_binary_cols(&mut EvalContext::default())?;
-            let retrieve_len = cols.len();
-            let mut cols_iter = cols.into_iter();
-            if self.col_len != retrieve_len {
-                if let Some(v) = cols_iter.next() {
-                    pk_builder.append(&v);
+        let mut is_drained = false;
+        while !is_drained {
+            let result = self.data.next_batch(BATCH_MAX_SIZE);
```

Comment:
Do we really need an incremental batch size style for analyze requests?

Reply:
In this code blocks, we just fetch a fixed number of rows from the `Storage`, which is `BATCH_MAX_SIZE == 1024`.